components/style/properties/longhand/position.mako.rs

```
@@ -220,3 +220,81 @@ ${helpers.single_keyword("object-fit", "fill contain cover none scale-down",
+            Column,
+        }
+
+        #[derive(PartialEq, Clone, Eq, Copy, Debug)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct T {
+            pub autoflow: AutoFlow,
+            pub dense: bool,
+        }
+    }
+
+    impl NoViewportPercentage for SpecifiedValue {}
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            dest.write_str(match self.autoflow {
+                computed_value::AutoFlow::Column => "column",
+                computed_value::AutoFlow::Row => "row"
+            })?;
+
+            if self.dense { dest.write_str(" dense")?; }
+            Ok(())
+        }
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T {
+            autoflow: computed_value::AutoFlow::Row,
+            dense: false
+        }
+    }
+
+    /// [ row | column ] || dense
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        let mut autoflow = false;
+        let mut value = get_initial_value();
+        loop {
+            match_ignore_ascii_case! { input.expect_ident()?,
+                "row" => if autoflow { break }
+                         else {
+                             autoflow = true;
+                             value.autoflow = computed_value::AutoFlow::Row;
+                             continue
+                         },
+                "column" => if autoflow { break }
+                            else {
+                                autoflow = true;
+                                value.autoflow = computed_value::AutoFlow::Column;
+                                continue
+                            },
+                "dense" => if value.dense { break }
+                           else { value.dense = true; continue },
+                _ => break
```

Similarly, this should return an error.